src/Fantomas/TokenParser.fs

```diff
@@ -260,15 +354,24 @@ let getDefineExprs sourceCode =
 
     result
 
-let getOptimizedDefinesSets sourceCode =
+let internal getOptimizedDefinesSets (hashTokens: Token list) =
     let maxSteps = FormatConfig.satSolveMaxStepsMaxSteps
 
-    match getDefineExprs sourceCode
+    match getDefineExprs hashTokens
           |> BoolExpr.mergeBoolExprs maxSteps
           |> List.map snd with
     | [] -> [ [] ]
     | xs -> xs
 
+let getDefines sourceCode =
+    let hashTokens = getTokenizedHashes sourceCode
+
+    let defineCombinations =
+        getOptimizedDefinesSets hashTokens @ (getDefinesWords  hashTokens |> List.map List.singleton) @ [ [] ]
```

Comment:
Hey, here's a formatting error (double space after `getDefinesWords`)! Sure we wanted that `.fantomasignore`? ;)